src/persist-client/src/usage.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Introspection of storage utilization by persist
+
+use mz_ore::metrics::MetricsRegistry;
+use std::sync::Arc;
+
+use mz_persist::cfg::BlobConfig;
+use mz_persist::location::{Blob, ExternalError};
+
+use crate::{retry_external, Metrics, ShardId};
+
+/// Provides access to storage usage metrics for a specific Blob
+#[derive(Debug)]
+pub struct StorageUsageClient {
+    blob: Arc<dyn Blob + Send + Sync>,
+}
+
+impl StorageUsageClient {
+    /// Creates a new StorageUsageClient pointed to a specific Blob
+    pub async fn open(blob_uri: &str) -> Result<StorageUsageClient, ExternalError> {
+        let blob = BlobConfig::try_from(blob_uri).await?;
+
+        // TODO: what do we want to do about the metrics here?
```

Comment:
wasn't sure about this -- `retry_external` currently requires metrics, but I didn't know how to ensure we have a metrics registry available to pass in since it's normally tied to the full persist client. alternatively, there could simply be a `retry_external` variant without metrics

Reply:
PersistClientCache has a copy and should be available where this usage stuff is called. let's somehow plumb that metrics here (maybe `open` takes a `&PersistClientCache`? haven't thought about the details much, also open to other ideas)

Reply:
Ooh if we have access to `PersistClientCache` that changes things for sure. I reworked this a bit and pulled out the `PersistClientCache`'s opening of a Blob into a `pub(crate)` method that `StorageUsageClient` calls into, so it will remain cached for any other callers.